webhook/webhook.go

```diff
@@ -40,37 +40,57 @@ func StartWebhook(ctx context.Context, webhookType string, clients *client.Clien
 	}()
 
 	logrus.Infof("Waiting for %v webhook to become ready", webhookType)
+	if !isServiceAvailable(webhookLocalEndpoint) {
+		return fmt.Errorf("%v webhook is not ready on localhost after %v sec", webhookType, defaultStartTimeout)
+	}
+	logrus.Infof("Started longhorn %s webhook server on localhost", webhookType)
+	return nil
+}
+
+func CheckWebhookServiceAvailability(webhookType string) error {
+	var webhookServiceEndpoint string
+	switch webhookType {
+	case types.WebhookTypeAdmission:
+		webhookServiceEndpoint = fmt.Sprintf("https://longhorn-admission-webhook.longhorn-system.svc:%d/v1/healthz", types.DefaultAdmissionWebhookPort)
+	case types.WebhookTypeConversion:
+		webhookServiceEndpoint = fmt.Sprintf("https://longhorn-conversion-webhook.longhorn-system.svc:%d/v1/healthz", types.DefaultConversionWebhookPort)
+	default:
+		return fmt.Errorf("unexpected webhook server type %v", webhookType)
+	}
+
+	if !isServiceAvailable(webhookServiceEndpoint) {
+		return fmt.Errorf("%v webhook service is not accessible after %v sec", webhookType, defaultStartTimeout)
```

Comment:
Suggest refactor `isServiceAvailable` with another parameter for `timeout` and pass `defaultStartTimeout` for it. This way will make the following log with `defaultStartTimeout` making more sense.

Reply:
Fixed.